dcpy/builds/load.py

```diff
@@ -0,0 +1,61 @@
+from pathlib import Path
+import typer
+
+from dcpy import REPO_ROOT_PATH
+from dcpy.utils import git, postgres
+from dcpy.utils.logging import logger
+from dcpy.connectors.edm import recipes
+
+
+def setup_environment(pg_client: postgres.PostgresClient):
+    pg_client.drop_schema()
+    pg_client.create_schema()
+    # TODO delete draft S3 folder
+
+
+def load_source_data(product: str):
+    product_path = REPO_ROOT_PATH / f"products/{product}"
+    build_name = git.run_name()
```

Comment:
Weren't we setting an environment var for the schema, rather than invoking git? e.g. the `BUILD_ENGINE_SCHEMA` in `knownprojects_build.yml`?

Reply:
**TLDR:** reading and [using](https://github.com/NYCPlanning/data-engineering/blob/edm-build-db/dcpy/utils/git.py#L60) an envar in python has been nicer and more maintainable than doing the same in bash in a github action

I guess asking _why_ we're invoking git leads me to wanting to do things in python

the why seems to be "we need to know the unique name for this draft build build/environment". and our chosen approach is to use the name of the branch or the PR number. since those are git-related values, we currently use `git.py`

and in the majority of cases, we're not invoking git. we're [reading an envar](https://github.com/NYCPlanning/data-engineering/blob/edm-build-db/dcpy/utils/git.py#L37-L41) which exists because it's a [github action default variable](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables)

for better separation, `build_name` could be determined and set to an envar well before something like `load_source_data` is called? at first I had separate `load.py` and `environemt.py` files but they only had 1 function in each

Reply:
> for better separation, build_name could be determined and set to an envar well before something like load_source_data is called?

I think this is the correct approach, with the goal of separating config from app code. Though, should we just use `BUILD_ENGINE_SCHEMA` for sake of consistency?

Reply:
sure could! the envar name might be too specific if it'd be used to pick an S3 folder, but when it's postgres-related, "schema" is on-point

Reply:
@alexrichey I went with [d658beb](https://github.com/NYCPlanning/data-engineering/pull/269/commits/d658beb11c898e29adf8d3b5e2d11a43076bfb36) to improve this

since AFIK, envars wouldn't persist across something like
```bash
python3 -m dcpy.builds.setup --product template
python3 -m dcpy.builds.load --product template
```
I thought it'd be nice to add `build_metadata.build_name` to use the `BUILD_ENGINE_SCHEMA` envar. and if the envar isn't set, do `git.run_name()`

if someday soon we can just do `python3 -m dcpy.builds.build --product template`, then maybe setting an envar would work. but I don't think we're there yet

Reply:
Cool - makes sense. For env var persistance, I think it would be nice to have a dcpy module that sets them. Maybe it would emit a list of keypairs, or write a `.env` file. Agree about just doing it later.